docs/SNICKER.md

```diff
@@ -0,0 +1,322 @@
+
+```
+2020-12-28 16:09:04,329 [INFO]  Finished processing block: 790
+2020-12-28 16:09:04,334 [INFO]  Found SNICKER transaction: 32f80807b3ba4ca477b25e8ab608a8a3134a34c8c3787cad95c653d1805d7533 in block: 791
+2020-12-28 16:09:04,338 [INFO]  Finished processing block: 791
+2020-12-28 16:09:04,340 [INFO]  Finished processing block: 792
+done
+```
+.. or similar. Then look in `./candidates.txt` to find the details of the identified transaction, including its full hex, which you need to copy:
+```
+2020-12-16 11:13:01,708 [INFO]  {
+    "hex": "0200000000010138e8a90b3df7740b9d5f5ae9af2cf6769f314d290b2e12bf25bfa4aae2c0cbe20000000000feffffff0280ba8c010000000016001471b09afbac6204627225c10f3a8d4a0749364fdb6d7c36220000000016001447ae59f32c504cbb56e18b77f7842fb58b55025b02473044022075354351ad4c619ba662f9abd25e8ee434f8381795001606a29fa959d36aeb7f022018f8bf1ec0407dad586baeb7e4d977aaacbd8fb15579293e6d739ad69ac3c6cf012103f8e827464fb83209c194376c53ae8f4e7ab5f1baf0948705fec6dd421f2b65c37a020000",
+    "inputs": [
+        .................
+Full transaction hex for creating a proposal is found in the above.
+The unspent indices are: 0 1 2
+```
+
+Copy that fully signed transaction hex, and note the unspent outputs. You're going to assume (in this case correctly of course) that all of the inputs are valid options for the SNICKER public key (why? because it was a *seed* transaction, it was a fake SNICKER, and so all the inputs belonged to the same wallet).
+
+At this point you're ready to run the proposal creator:
+
+```
+python create-encrypted-proposal.py --datadir=. proposerwallet.jmdat "0200000000010138e8a90b3df7740b9d5f5ae9af2cf6769f314d290b2e12bf25bfa4aae2c0cbe20000000000feffffff0280ba8c010000000016001471b09afbac6204627225c10f3a8d4a0749364fdb6d7c36220000000016001447ae59f32c504cbb56e18b77f7842fb58b55025b02473044022075354351ad4c619ba662f9abd25e8ee434f8381795001606a29fa959d36aeb7f022018f8bf1ec0407dad586baeb7e4d977aaacbd8fb15579293e6d739ad69ac3c6cf012103f8e827464fb83209c194376c53ae8f4e7ab5f1baf0948705fec6dd421f2b65c37a020000" 0 1 100 -m1 -n
+```
+Obviously see the `--help` for details, but in this example we chose input index 0 to source the pubkey for the encryption, output index 1 for the coin we want the receiver to spend with us in the coinjoin, 100 sats as the amount we will bump their output by as an incentive to do the coinjoin (you *can* make this number negative, to receive, instead - the proposer is paying the tx fee otherwise, note), and mixdepth 1 as the mixdepth from which we source *our* coins for the coinjoin. Make sure of course that mixdepth 1 has at least a little bit more bitcoin than the size of that output at index 1 aforementioned. The `-n` option can be used to output the proposal to stdout (it is base64+hex so copy-pasteable).
```

Comment:
Here too a list could come in handy:

---

Obviously see the `--help` for details, but in this example we chose:

* Input index 0 to source the pubkey for the encryption.
* Output index 1 for the coin we want the receiver to spend with us in the coinjoin.
* 100 sats as the amount we will bump their output by as an incentive to do the coinjoin (you *can* make this number negative, to receive, instead - the proposer is paying the tx fee otherwise, note).
* Mixdepth 1 as the mixdepth from which we source *our* coins for the coinjoin. Make sure of course that mixdepth 1 has at least a little bit more bitcoin than the size of that output at index 1 aforementioned.
* The `-n` option can be used to output the proposal to stdout (it is base64+hex so copy-pasteable).

---

Also, `you *can* make this number negative, to receive, instead - the proposer is paying the tx fee otherwise, note` could be rephrased as to me seems a bit confusing.

Reply:
Agree on this one